### DISCUSSION TITLE
Difference between #[server] and #[server(MyFunction)] for fullstack app?

### DISCUSSION THREAD

**Discussion by ThomasCartier** [DioxusLabs/dioxus#2726] [Q&A] [answered]
Hi,
My fullstack apps run with both 

```
#[server(GetServerData)]
async fn get_server_data() -> Result<String, ServerFnError> {
    Ok("Hello from the server!".to_string())
}
```

and 
```
#[server]
async fn get_server_data() -> Result<String, ServerFnError> {
    Ok("Hello from the server!".to_string())
}

```

Just fine using the latest Dioxus.
Is there a real need for the first version ? It looks registered on the server side either way.

Thanks

**Comment by ealmloff** [comment, marked as answer]
`#[server]` expands to `#[server(GetServerData)]` with a name generated from the upper camel case version of the function name. You rarely need to know the name of the struct that implements the server function trait so `#[server]` is generally fine
